### ISSUE TITLE
Test suite fails to run with 1.0.10rc1

### ISSUE THREAD

**Issue by lfam** [borgbackup/borg#2112]
When testing 1.0.10rc1 with GNU Guix, pytest collects no tests and so the test suite fails, like this:

```
starting phase `check'
============================= test session starts ==============================
platform linux -- Python 3.5.2, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /gnu/store/wxpb93hg5j8ydmmg35lgjd2sxpvwbym3-python-wrapper-3.5.2/bin/python
cachedir: ../.cache
rootdir: /tmp/guix-build-borg-1.0.10rc1.drv-0/borgbackup-1.0.10rc1, inifile: 
collecting ... collected 0 items

========================= no tests ran in 0.02 seconds =========================
phase `check' failed after 0.5 seconds
```

With borg 1.0.9 [0], the test suite runs and passes its tests. I verified that the test suite runs successfully from the latest commit on the Guix master branch. The only difference in the failing case is the new version of borg.

Please let me know what other information I can provide.

[0]
http://git.savannah.gnu.org/cgit/guix.git/tree/gnu/packages/backup.scm?id=cf95bd3a8dc1440dbb9788e96f726bcdf54f22f6#n445

**Comment by ThomasWaldmann** [comment]
I invoked line 452+ (without the long -k arg) with 1.0.10rc1 source manually on a shell and it worked for me.

**Comment by ThomasWaldmann** [comment]
Hmm, maybe changing line 1 in 1.0-maint was a bad idea.

Shall I take that back in 1.0-maint (and master?) so pytest < 3.0 will still work?

**Comment by lfam** [comment]
If the issue is the pytest version, the solution is your choice. I already created a special pytest package for borg 1.0.8. I can update it to 3.0.x if necessary.

I'll try with 3.0.x later today and report back.

**Comment by ThomasWaldmann** [comment]
@LocutusOfBorg how about debian/ubuntu, guess it is better not to require pytest 3.0, right?

**Comment by enkore** [comment]
IIRC it is built against jessie-backports, which has 3.0.3: https://packages.debian.org/jessie-backports/python3-pytest

Anyhow it's probably better to avoid a major version dependency chain in a point release.

**Comment by LocutusOfBorg** [comment]
Debian is fine that has 3.0.3 in backports, ubuntu has still 2.x (3.0 is not migrating due to a qtpy regression)

**Comment by ThomasWaldmann** [comment]
OK, I'll revert that change. It was just to get rid of a deprecation warning in pytest output.

**Comment by LocutusOfBorg** [comment]
I made pytest 3.0.6 migrate on Ubuntu zesty a few hours ago

**Comment by bpereto** [comment]
Fedora 25 and Epel 7 have also pytest < 3.0.0

The newer Fedora Rawhide have 3.0.6.

**Comment by enkore** [comment]
This should be fixed by #2122.
